src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/AlertCodesController.kt

```diff
@@ -107,6 +109,75 @@ class AlertCodesController(
     httpRequest: HttpServletRequest,
   ) = alertCodeService.deactivateAlertCode(alertCode, httpRequest.alertRequestContext())
 
+  @PreAuthorize("hasAnyRole('$ROLE_ALERTS_ADMIN')")
+  @GetMapping
+  @Operation(
+    summary = "Retrieve all alert codes",
+    description = "Retrieve all alert codes, typically from the Alerts UI",
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Alert code retrieved",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
```

Comment:
I don't think this function can return a 404 as it's getting all codes?

Reply:
Done.